ledger/src/blockstore.rs

```diff
@@ -3594,37 +3598,35 @@ impl Blockstore {
         // `consumed` is the next missing shred index, but shred `i` existing in
         // completed_data_end_indexes implies it's not missing
         assert!(!completed_data_indexes.contains(&consumed));
-        completed_data_indexes
-            .range(start_index..consumed)
-            .scan(start_index, |begin, index| {
-                let out = (*begin, *index);
-                *begin = index + 1;
-                Some(out)
-            })
+        std::iter::once(start_index)
+            .chain(
+                completed_data_indexes
+                    .range(start_index..consumed)
+                    .map(|index| index + 1),
+            )
```

Comment:
This part was a little messy, it seems `tuple_windows` could clean this up if we retain the original return type